### PR TITLE
*: add tidb_ddl_distribute_reorg item to system-variables

### PR DESCRIPTION
**Pull request by zimulala**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)
Add `tidb_ddl_distribute_reorg` to system-variables.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v6.6 (TiDB 6.6 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.4 (TiDB 6.4 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s): https://github.com/pingcap/tidb/pull/40542

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Comment by zimulala** [comment]
PTAL @Benjamin2037 @Defined2014

File: system-variables.md [hunk 1]
```diff
@@ -956,6 +956,19 @@ MPP 是 TiFlash 引擎提供的分布式计算框架，允许节点之间的数
...
+
+- 作用域：GLOBAL
+- 是否持久化到集群：是
+- 默认值：`OFF`
+- 这个变量用来控制是否开启分布式执行 DDL reorg 阶段，来提升此阶段的速度。目前此开关只对 `ADD INDEX` 语句有效。开启该变量对于数据量较大的表有一定的性能提升。
+- 要验证已经完成的 `ADD INDEX` 操作是否使用了此功能，可以查看 `tidb_ddl_backfill_history` 表是否有对应任务。
+
+> **警告：**
+>
+> 当前该功能为实验特性，不建议在生产环境中使用。
```
Comment (zimulala) [inline system-variables.md:969]:
```suggestion
> 当前该功能为实验特性，不建议在生产环境中使用。
> 当前启用此功能后，在 DDL reorg 阶段遇到某些异常只会做简单重试，还没有兼容 DDL 的重试方式（即依据  `tidb_ddl_error_count_limit` 的大小控制重试次数）。
```

File: system-variables.md [hunk 2]
```diff
@@ -956,6 +956,20 @@ MPP 是 TiFlash 引擎提供的分布式计算框架，允许节点之间的数
 > - 如果先启动添加索引加速任务，再启动 PITR 备份任务，此时 PITR 备份任务会报错，但不影响正在添加索引的任务。
 > - 如果同时启动 PITR 备份任务和添加索引加速任务，可能会由于两个任务无法察觉到对方而导致 PITR 不能成功备份增加的索引数据。
 
+### `tidb_ddl_distribute_reorg` <span class="version-mark">从 v6.6.0 版本开始引入</span>
+
+- 作用域：GLOBAL
+- 是否持久化到集群：是
+- 默认值：`OFF`
+- 这个变量用来控制是否开启分布式执行 DDL reorg 阶段，来提升此阶段的速度。目前此开关只对 `ADD INDEX` 语句有效。开启该变量对于数据量较大的表有一定的性能提升。
+- 要验证已经完成的 `ADD INDEX` 操作是否使用了此功能，可以查看 `tidb_ddl_backfill_history` 表是否有对应任务。
```
Comment (Defined2014) [inline system-variables.md:965]:
```suggestion
- 要验证已经完成的 `ADD INDEX` 操作是否使用了此功能，可以查看 `mysql.tidb_ddl_backfill_history` 表是否有对应任务。
```

**Review by Defined2014 — APPROVED** [review]

File: system-variables.md [hunk 3]
```diff
@@ -956,6 +956,20 @@ MPP 是 TiFlash 引擎提供的分布式计算框架，允许节点之间的数
...
+
+- 作用域：GLOBAL
+- 是否持久化到集群：是
+- 默认值：`OFF`
+- 这个变量用来控制是否开启分布式执行 DDL reorg 阶段，来提升此阶段的速度。目前此开关只对 `ADD INDEX` 语句有效。开启该变量对于数据量较大的表有一定的性能提升。
+- 要验证已经完成的 `ADD INDEX` 操作是否使用了此功能，可以查看 `mysql.tidb_ddl_backfill_history` 表是否有对应任务。
+
+> **警告：**
+>
+> 当前该功能为实验特性，不建议在生产环境中使用。
```
Comment (Benjamin2037) [inline system-variables.md:969]:
```suggestion
> 当前该功能为实验特性
```

File: system-variables.md [hunk 4]
```diff
@@ -956,6 +956,20 @@ MPP 是 TiFlash 引擎提供的分布式计算框架，允许节点之间的数
 > - 如果先启动添加索引加速任务，再启动 PITR 备份任务，此时 PITR 备份任务会报错，但不影响正在添加索引的任务。
 > - 如果同时启动 PITR 备份任务和添加索引加速任务，可能会由于两个任务无法察觉到对方而导致 PITR 不能成功备份增加的索引数据。
 
+### `tidb_ddl_distribute_reorg` <span class="version-mark">从 v6.6.0 版本开始引入</span>
+
+- 作用域：GLOBAL
+- 是否持久化到集群：是
+- 默认值：`OFF`
+- 这个变量用来控制是否开启分布式执行 DDL reorg 阶段，来提升此阶段的速度。目前此开关只对 `ADD INDEX` 语句有效。开启该变量对于数据量较大的表有一定的性能提升。
+- 要验证已经完成的 `ADD INDEX` 操作是否使用了此功能，可以查看 `mysql.tidb_ddl_backfill_history` 表是否有对应任务。
```
Comment (Benjamin2037) [inline system-variables.md:965]: 这个后续是不是表名是不是会改为通用的？还是这个是我们 DDL 特有的？

Comment (zimulala) [reply system-variables.md:965]: 确实，或者改了表名在提个 PR？目前这个名字还不一定。

File: system-variables.md [hunk 5]
```diff
@@ -956,6 +956,20 @@ MPP 是 TiFlash 引擎提供的分布式计算框架，允许节点之间的数
 > - 如果先启动添加索引加速任务，再启动 PITR 备份任务，此时 PITR 备份任务会报错，但不影响正在添加索引的任务。
 > - 如果同时启动 PITR 备份任务和添加索引加速任务，可能会由于两个任务无法察觉到对方而导致 PITR 不能成功备份增加的索引数据。
 
+### `tidb_ddl_distribute_reorg` <span class="version-mark">从 v6.6.0 版本开始引入</span>
+
+- 作用域：GLOBAL
+- 是否持久化到集群：是
+- 默认值：`OFF`
+- 这个变量用来控制是否开启分布式执行 DDL reorg 阶段，来提升此阶段的速度。目前此开关只对 `ADD INDEX` 语句有效。开启该变量对于数据量较大的表有一定的性能提升。
```
Comment (zimulala) [inline system-variables.md:964]:
```suggestion
- 这个变量用于控制是否开启分布式执行 DDL reorg 阶段，来提升此阶段的速度。目前此开关只对 `ADD INDEX` 语句有效。开启该变量对于数据量较大的表有一定的性能提升。
```

**Review by Benjamin2037 — APPROVED** [review]
LGTM

File: system-variables.md [hunk 6]
```diff
@@ -956,6 +956,20 @@ MPP 是 TiFlash 引擎提供的分布式计算框架，允许节点之间的数
 > - 如果先启动添加索引加速任务，再启动 PITR 备份任务，此时 PITR 备份任务会报错，但不影响正在添加索引的任务。
 > - 如果同时启动 PITR 备份任务和添加索引加速任务，可能会由于两个任务无法察觉到对方而导致 PITR 不能成功备份增加的索引数据。
 
+### `tidb_ddl_distribute_reorg` <span class="version-mark">从 v6.6.0 版本开始引入</span>
+
```
Comment (ran-huang) [inline system-variables.md:960]:
```suggestion

> **Warning:**
>
> 该功能目前为实验特性。
> 当前启用此功能后，在 DDL reorg 阶段遇到某些异常只会做简单重试，还没有兼容 DDL 的重试方式，即目前无法依据  [`tidb_ddl_error_count_limit`](#tidb_ddl_error_count_limit) 的大小控制重试次数。

```

File: system-variables.md [hunk 7]
```diff
@@ -956,6 +956,20 @@ MPP 是 TiFlash 引擎提供的分布式计算框架，允许节点之间的数
...
+- 默认值：`OFF`
+- 这个变量用于控制是否开启分布式执行 DDL reorg 阶段，来提升此阶段的速度。目前此开关只对 `ADD INDEX` 语句有效。开启该变量对于数据量较大的表有一定的性能提升。
+- 要验证已经完成的 `ADD INDEX` 操作是否使用了此功能，可以查看 `mysql.tidb_ddl_backfill_history` 表是否有对应任务。
+
+> **警告：**
+>
+> 当前该功能为实验特性
+> 当前启用此功能后，在 DDL reorg 阶段遇到某些异常只会做简单重试，还没有兼容 DDL 的重试方式（即依据  `tidb_ddl_error_count_limit` 的大小控制重试次数）。
+
+
```
Comment (ran-huang) [inline system-variables.md:972]:
```suggestion
```

File: system-variables.md [hunk 8]
```diff
@@ -956,6 +956,20 @@ MPP 是 TiFlash 引擎提供的分布式计算框架，允许节点之间的数
 > - 如果先启动添加索引加速任务，再启动 PITR 备份任务，此时 PITR 备份任务会报错，但不影响正在添加索引的任务。
 > - 如果同时启动 PITR 备份任务和添加索引加速任务，可能会由于两个任务无法察觉到对方而导致 PITR 不能成功备份增加的索引数据。
 
+### `tidb_ddl_distribute_reorg` <span class="version-mark">从 v6.6.0 版本开始引入</span>
```
Comment (ran-huang) [inline system-variables.md:959]: @hawkingrei 麻烦看看，要不要提一下DDL 动态资源管控

File: system-variables.md [hunk 9]
```diff
@@ -956,6 +956,19 @@ MPP 是 TiFlash 引擎提供的分布式计算框架，允许节点之间的数
...
+
+> **Warning:**
+>
+> 该功能目前为实验特性。
+> 当前启用此功能后，在 DDL reorg 阶段遇到某些异常只会做简单重试，还没有兼容 DDL 的重试方式，即目前无法依据  [`tidb_ddl_error_count_limit`](#tidb_ddl_error_count_limit) 的大小控制重试次数。
+
+- 作用域：GLOBAL
+- 是否持久化到集群：是
+- 默认值：`OFF`
+- 这个变量用于控制是否开启分布式执行 DDL reorg 阶段，来提升此阶段的速度。目前此开关只对 `ADD INDEX` 语句有效。开启该变量对于数据量较大的表有一定的性能提升。
```
Comment (ran-huang) [inline system-variables.md:969]: @zimulala PTAL
```suggestion
- 这个变量用于控制是否开启分布式执行 DDL reorg 阶段，来提升此阶段的速度。目前此开关只对 `ADD INDEX` 语句有效。开启该变量对于数据量较大的表有一定的性能提升。分布式 DDL 会通过 DDL 动态资源管控，控制 DDL 的 CPU 使用量，来防止对线上业务产生影响。
```

Comment (zimulala) [reply system-variables.md:969]: LGTM

**Review by ran-huang — APPROVED** [review]

File: system-variables.md [hunk 10]
```diff
@@ -956,6 +956,19 @@ MPP 是 TiFlash 引擎提供的分布式计算框架，允许节点之间的数
 > - 如果先启动添加索引加速任务，再启动 PITR 备份任务，此时 PITR 备份任务会报错，但不影响正在添加索引的任务。
 > - 如果同时启动 PITR 备份任务和添加索引加速任务，可能会由于两个任务无法察觉到对方而导致 PITR 不能成功备份增加的索引数据。
 
+### `tidb_ddl_distribute_reorg` <span class="version-mark">从 v6.6.0 版本开始引入</span>
+
+> **Warning:**
+>
+> 该功能目前为实验特性。
+> 当前启用此功能后，在 DDL reorg 阶段遇到某些异常只会做简单重试，还没有兼容 DDL 的重试方式，即目前无法依据  [`tidb_ddl_error_count_limit`](#tidb_ddl_error_count_limit) 的大小控制重试次数。
```
Comment (ran-huang) [inline system-variables.md:964]:
```suggestion
> **警告：**
>
> - 该功能目前为实验特性。
> - 当前启用此功能后，在 DDL reorg 阶段遇到某些异常只会做简单重试，还没有兼容 DDL 的重试方式，即目前无法依据  [`tidb_ddl_error_count_limit`](#tidb_ddl_error_count_limit) 的大小控制重试次数。
```

File: system-variables.md [hunk 11]
```diff
@@ -956,6 +956,19 @@ MPP 是 TiFlash 引擎提供的分布式计算框架，允许节点之间的数
 > - 如果先启动添加索引加速任务，再启动 PITR 备份任务，此时 PITR 备份任务会报错，但不影响正在添加索引的任务。
 > - 如果同时启动 PITR 备份任务和添加索引加速任务，可能会由于两个任务无法察觉到对方而导致 PITR 不能成功备份增加的索引数据。
 
+### `tidb_ddl_distribute_reorg` <span class="version-mark">从 v6.6.0 版本开始引入</span>
+
+> **警告：**
+>
+> - 该功能目前为实验特性。
```
Comment (TomShawn) [inline system-variables.md:963]:
```suggestion
> - 该功能目前为实验特性。不推荐在生产环境中开启该变量。
```

File: system-variables.md [hunk 12]
```diff
@@ -956,6 +956,19 @@ MPP 是 TiFlash 引擎提供的分布式计算框架，允许节点之间的数
...
+> - 当前启用此功能后，在 DDL reorg 阶段遇到某些异常只会做简单重试，还没有兼容 DDL 的重试方式，即目前无法依据  [`tidb_ddl_error_count_limit`](#tidb_ddl_error_count_limit) 的大小控制重试次数。
```
Comment (TomShawn) [inline system-variables.md:964]:
```suggestion
> - 当前启用此功能后，在 DDL reorg 阶段遇到某些异常只会做简单重试，还没有兼容 DDL 操作的重试方式，即目前无法依据  [`tidb_ddl_error_count_limit`](#tidb_ddl_error_count_limit) 的大小控制重试次数。
```

**Review by TomShawn — APPROVED** [review]
rest LGTM

**Pull request merged**